From 512dc88fb034a2e7873acd0bb8765e198a5d7d2d Mon Sep 17 00:00:00 2001
From: Florian Westphal <fwestpha@redhat.com>
Date: Sun, 14 Sep 2014 19:49:51 -0400
Subject: [net] Allocate a new 16 bits for flags in skbuff

Message-id: <1410724197-31885-2-git-send-email-fwestpha@redhat.com>
Patchwork-id: 94692
O-Subject: [PATCH 46/57 RHEL7.1 net v2] net: Allocate a new 16 bits for flags in skbuff
Bugzilla: 1131999
RH-Acked-by: Jiri Benc <jbenc@redhat.com>
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>

BZ: 1131999
Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=7951596

KABI breaker.
Move it closer to the end and insert it right before first
rhel specific padding member, closing a 16 bit padding hole.

commit de20fe8e2cc3c4ca13fdb529e6720d9d199333fe
Author: Tom Herbert <therbert@google.com>
Date:   Wed Aug 27 21:26:35 2014 -0700

    net: Allocate a new 16 bits for flags in skbuff

    Signed-off-by: Tom Herbert <therbert@google.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index a1bfcc9..3518bd6 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -529,6 +529,11 @@ struct sk_buff {
  __u16   network_header;
  __u16   mac_header;
 
+#ifndef __GENKSYMS__
+ kmemcheck_bitfield_begin(flags3);
+ /* 16 bit hole */
+ kmemcheck_bitfield_end(flags3);
+#endif
  /* RHEL SPECIFIC
   *
   * The following padding has been inserted before ABI freeze to
-- 
1.7.1